keys.c

```diff
@@ -125,6 +125,7 @@ new_Public_Key_Packet(int len)
 		break;
 	case 18:/*ECDH*/
 		oidLEN = Getc();
+		memset(oid_input_HEX,0,10);
```

Comment:
I would like to avoid hard-coding a magic number. Could you use `oid_input_HEX_size` instead of `10`?